modules/abstract-eth/src/EthLikeToken.ts

```diff
@@ -0,0 +1,112 @@
+/**
+ * @prettier
+ */
+import { coins, EthLikeTokenConfig, tokens } from '@bitgo/statics';
+
+import { BitGoBase, CoinConstructor, NamedCoinConstructor } from '@bitgo/sdk-core';
+import { AbstractEthLikeCoin } from './abstractEthLikeCoin';
+import { TransactionBuilder as EthTransactionBuilder, TransactionPrebuild } from '@bitgo/sdk-coin-eth';
+
+export type CoinNames = {
+  [network: string]: string;
+};
+
+export class EthLikeToken extends AbstractEthLikeCoin {
+  public readonly tokenConfig: EthLikeTokenConfig;
+
+  protected constructor(bitgo: BitGoBase, tokenConfig: EthLikeTokenConfig, coinNames: CoinNames) {
+    const staticsCoin = coins.get(coinNames[tokenConfig.network]);
+    super(bitgo, staticsCoin);
+    this.tokenConfig = tokenConfig;
+  }
+
+  static createTokenConstructor(config: EthLikeTokenConfig, coinNames: CoinNames): CoinConstructor {
+    return (bitgo: BitGoBase) => new this(bitgo, config, coinNames);
+  }
+
+  static createTokenConstructors(coinNames: CoinNames): NamedCoinConstructor[] {
+    const tokensCtors: NamedCoinConstructor[] = [];
+    const chain = coinNames.Mainnet;
```

Comment:
would this work? or are tokens.testnet indexed by mainnet token names?

Reply:
So the code before just hardcoded `chain` to be `bsc` (in the case of binance tokens), so I assumed that we always use the mainnet chain name. It seems that the testnet tokens are in `...tokens.testnet[chain].tokens]` and mainnet tokens are in `...tokens.bitcoin[chain].tokens]`